### ISSUE TITLE
Exception question

### ISSUE THREAD

**Issue by higgsja** [studerw/td-ameritrade-client#29]
I really like what you have done with this.

I've been running some option data retrievals for several months. Yesterday, I started getting an exception that I cannot understand. Can you take a look?

Here is the console output:
`08:26:28.555 [main] INFO com.studerw.tda.client.HttpTdaClient - Initiating HttpTdaClient...
08:26:28.903 [main] INFO com.studerw.tda.client.HttpTdaClient - get option chain for symbol: VXX
08:26:28.961 [main] INFO TDA_HTTP - REQUEST GET: https://api.tdameritrade.com/v1/marketdata/chains?symbol=VXX
08:26:28.961 [main] DEBUG TDA_HTTP - REQUEST Headers:
 Accept: application/json
Authorization: Bearer UNSET

Exception in thread "main" java.lang.RuntimeException: java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_CBC_SHA, SSL_RSA_WITH_3DES_EDE_CBC_SHA], tlsVersions=[TLS_1_3, TLS_1_2], supportsTlsExtensions=true), ConnectionSpec()], supported protocols=[TLSv1]
	at com.studerw.tda.client.HttpTdaClient.getOptionChain(HttpTdaClient.java:589)
	at com.hpi.appcontrollers.TDAmeritradeFetchOptionsController.doPrices(TDAmeritradeFetchOptionsController.java:37)
	at com.hpi.appcontrollers.CmdLineController.doCommandLine(CmdLineController.java:224)
	at com.hpi.appcontrollers.AppController.startApp(AppController.java:62)
	at com.hpi.appTPCcli.TPCcliEx.main(TPCcliEx.java:19)
Caused by: java.net.UnknownServiceException: Unable to find acceptable protocols. isFallback=false, modes=[ConnectionSpec(cipherSuites=[TLS_AES_128_GCM_SHA256, TLS_AES_256_GCM_SHA384, TLS_CHACHA20_POLY1305_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_256_CBC_SHA, SSL_RSA_WITH_3DES_EDE_CBC_SHA], tlsVersions=[TLS_1_3, TLS_1_2], supportsTlsExtensions=true), ConnectionSpec()], supported protocols=[TLSv1]
	at okhttp3.internal.connection.ConnectionSpecSelector.configureSecureSocket(ConnectionSpecSelector.kt:63)
	at okhttp3.internal.connection.RealConnection.connectTls(RealConnection.kt:345)
	at okhttp3.internal.connection.RealConnection.establishProtocol(RealConnection.kt:310)
	at okhttp3.internal.connection.RealConnection.connect(RealConnection.kt:178)
	at okhttp3.internal.connection.ExchangeFinder.findConnection(ExchangeFinder.kt:236)
	at okhttp3.internal.connection.ExchangeFinder.findHealthyConnection(ExchangeFinder.kt:109)
	at okhttp3.internal.connection.ExchangeFinder.find(ExchangeFinder.kt:77)
	at okhttp3.internal.connection.Transmitter.newExchange$okhttp(Transmitter.kt:162)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.kt:35)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.kt:82)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.kt:84)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.kt:71)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at com.studerw.tda.http.LoggingInterceptor.intercept(LoggingInterceptor.java:57)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at com.studerw.tda.client.OauthInterceptor.intercept(OauthInterceptor.java:46)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:112)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.kt:87)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.kt:184)
	at okhttp3.RealCall.execute(RealCall.kt:66)
	at com.studerw.tda.client.HttpTdaClient.getOptionChain(HttpTdaClient.java:585)
	... 4 more
`

**Comment by studerw** [comment]
Hi @higgsja - Sorry been busy for a week. Was this problem ever resolved? I've never seen an SSL error from TDA, though maybe they changed their API server certs...

**Comment by higgsja** [comment]
Got the error for about a week then it resolved itself. Strange.

**Comment by higgsja** [comment]
Finally think I resolved the http issue. You are using different versions of  okio and okhttp3 than I had; switched to what you are using and that went away.

I'm left with the same issues as #31 on option chains. Tracked it down to 
`DefaultMapper
	public static <T> T fromJson(InputStream in, TypeReference<T> typeReference) {
    try (BufferedInputStream ignored = new BufferedInputStream(in)) {
      return defaultMapper.readValue(in, typeReference);
    } catch (IOException e) {
      throw new RuntimeException(e);
    }`
with an exception like:
`Cannot deserialize value of type `java.math.BigDecimal` from String "NaN": not a valid representation at [Source: (BufferedInputStream); line: 1, column: 38443] (through reference chain: com.studerw.tda.model.option.OptionChain["putExpDateMap"]->java.util.LinkedHashMap["2020-10-23:5"]->java.util.LinkedHashMap["42.0"]->java.util.ArrayList[0]`

Apparently some field on the return is not a number but I am not smart enough to track further. I've tried a couple tickers, same result.

I'll also post this under #31.
